task2/task2_solution.py

```diff
@@ -0,0 +1,110 @@
+# NOTES:
+#
+# no idea whether it's ok to make
+# to_letters function a method of the class
+# (staticmethod doesn't work)
+# external func works fine, let is be so
+#
+# besides that, Journal class feels to be
+# not really universal (adding more fields to
+# input data requires changes in code)
+
+
+def to_letters(grades):
+    res = ""
+    for grade in grades:
+        if grade >= 95.0:
+            res += "A"
+        elif grade >= 85.0:
+            res += "B"
+        elif grade >= 75.0:
+            res += "C"
+        elif grade >= 65.0:
+            res += "D"
+        elif grade >= 60.0:
+            res += "E"
+        elif grade < 60.0:
+            res += "F"
+        res += " "
+    return res.strip()
+
+
+# calc overall average grade
+def avg_grade(homework, quizzes, tests):
+    grades = homework + quizzes + tests
+    res = sum(grades)/len(grades)
+    return res
+
+
+class Journal:
+    student_data = []
+
+    def __init__(self, students):
+        for dict in students:
```

Comment:
There is no need to go over elements of list and append them to another list
It's the same as just assigning old list to new